OnboardPietroPieri/App.tsx

```diff
@@ -5,23 +5,138 @@
  * @format
  */
 
-import React from "react";
-import {View, Text, StyleSheet} from 'react-native';
+import React, {PropsWithChildren, useState} from "react";
+import {
+  Alert,
+  TouchableOpacity,
+  TextInput,
+  SafeAreaView,
+  View,
+  Text,
+  StyleSheet
+} from 'react-native';
 
 const styles = StyleSheet.create({
+  safeArea: {
+    marginTop: 80,
+    flex: 1,
+  },
   center: {
+    marginTop: 30,
     alignItems: 'center',
-    flex: 1,
     justifyContent: 'center',
+    flex: 0.3,
+    paddingHorizontal: 24,
+  },
+  header: {
+    fontSize: 30,
+    fontWeight: 'bold',
+    textAlign: 'center',
+    color: 'black',
+  },
+  inputText: {
+    color: 'black',
+    fontWeight: 'normal',
+    marginLeft: 40,
+    top: 28,
+  },
+  input: {
+    marginLeft: 40,
+    marginTop: 30,
+    height: 50,
+    borderColor: 'gray',
+    borderWidth: 2,
+    paddingHorizontal: 10,
+    width: '80%',
+    borderRadius: 10,
   },
+  button: {
+    height: 50,
+    marginTop: 50,
+    marginLeft: 40,
+    backgroundColor: '#9403fc',
+    padding: 10,
+    borderRadius: 10,
+    width: '80%',
+  },
+  buttonText: {
+    textAlign: 'center',
+    fontWeight: 'bold',
+    color: 'white',
+    fontSize: 16,
+  }
+  
+
 });
 
-const App = () => {
+type SectionProps = PropsWithChildren<{
+  type?: string,
+  title: string;
+  setLogin: React.Dispatch<React.SetStateAction<string>>;
+}>;
+
+
+
+function Section({title, setLogin, type}: SectionProps): React.JSX.Element {
+  const isPassword = type === 'password';
+
   return (
-    <View style={styles.center}>
-      <Text>"Hello word"</Text>
+    <View>
+      <Text style={styles.inputText}>{title}</Text>
+      <TextInput
+        style={styles.input}
+        keyboardType={isPassword ? 'default' : 'email-address'}
+        autoCapitalize="none"
+        secureTextEntry={isPassword}
+        onChangeText={(text: React.SetStateAction<string>) => setLogin(text)}>
```

Comment:
no onChangeText o text é do tipo string e não `SetStateAction<string>`, esse `SetStateAction` é mapeado em `S | ((prevState: S) => S)` então a ideia é que não precisa de nada além de `string` nesse caso
```suggestion
        onChangeText={(text: string) => setLogin(text)}>
```

Reply:
aplicado